### ISSUE TITLE
Use tensorflow function inside keras.

### ISSUE THREAD

**Issue by kris-singh** [keras-team/keras#9818]
I want to do sampling based on the loss of the data points. For this i need to compute the loss for each data point. categorical_cross_entropy(y_true, y_pred) requires y_true to be tensor. So i just used tf.nn.softmax_cross_entropy_with_logits(Here is the gist https://gist.github.com/kris-singh/53b1c6c687ce2b44c0140f87628ea370.) but this creates a node in the graph again and again(i guess) which is the reason the program is very slow. How should i convert the code. If i define this in the __init__ function and use sess.run(res, feed_dict)(https://gist.github.com/kris-singh/364867d41d4c65fe356ac3d454fba596) which sess variable should i use. 

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by gabrieldemarmiesse** [comment]
I don't know if I understand correctly, but what you could use is K.function in the `__init__()` to convert tf.nn.softmax_cross_entropy_with_logits into a function that takes numpy arrays as inputs. Once this function is created, you can use it over and over and it won't create more nodes.

example:

```python
a = tf.placeholder(shape=(1,), dtype=tf.float32)
b = tf.placeholder(shape=(1,), dtype=tf.float32)
c = a + b +1
np_function = K.function([a, b], [c])

array1 = np.array([12], dtype=np.float32)
array2 = np.array([17], dtype=np.float32)

print(np_function([array1, array2]))
print(np_function([array1, array2]))
```
30
30

and no new nodes at each function call!

**Comment by kris-singh** [comment]
Thanks this solved the issue.
